client/service/utils/query.go

```diff
@@ -0,0 +1,186 @@
+package utils
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/irisnet/irishub/app/protocol"
+	"github.com/irisnet/irishub/app/v3/service"
+	"github.com/irisnet/irishub/client/context"
+	"github.com/irisnet/irishub/client/tendermint/tx"
+	sdk "github.com/irisnet/irishub/types"
+)
+
+// QueryRequestContext query a single request context
+func QueryRequestContext(cliCtx context.CLIContext, params service.QueryRequestContextParams) (
+	requestContext service.RequestContext, err error) {
+	bz, err := cliCtx.Codec.MarshalJSON(params)
+	if err != nil {
+		return requestContext, err
+	}
+	route := fmt.Sprintf("custom/%s/%s", protocol.ServiceRoute, service.QueryRequestContext)
+	res, err := cliCtx.QueryWithData(route, bz)
+	if err != nil {
+		return requestContext, err
+	}
+
+	_ = cliCtx.Codec.UnmarshalJSON(res, &requestContext)
+	if requestContext.Empty() {
+		requestContext, err = QueryRequestContextByTxQuery(cliCtx, params)
+		if err != nil {
+			return requestContext, err
+		}
+	}
+
+	if requestContext.Empty() {
+		return requestContext, fmt.Errorf("unknown request context: %s", hex.EncodeToString(params.RequestContextID))
+	}
```

Comment:
QueryRequestContext method parameter `requestContext` is better to use a pointer?